e2e/playwright/actions/src/tests/library/library-actions.spec.ts

```diff
@@ -0,0 +1,445 @@
+  const user1Library2 = `playwright-U1-library-${Utils.random()}`;
+  const user1Library3 = `playwright-U1-library-${Utils.random()}`;
+  const user1Library4 = `playwright-U1-library-${Utils.random()}`;
+  const user1Library5 = `playwright-U1-library-${Utils.random()}`;
+  const user2Library1 = `playwright-U2-library-${Utils.random()}`;
+  const user2Library2 = `playwright-U2-library-${Utils.random()}`;
+  const user2Library3 = `playwright-U2-library-${Utils.random()}`;
+  const user2Library4 = `playwright-U2-library-${Utils.random()}`;
+  const user2Library5Delete = `playwright-T-library-${Utils.random()}`;
+  const user2Library6Delete = `playwright-T-library-${Utils.random()}`;
+  const user2Library7Delete = `playwright-T-library-${Utils.random()}`;
+
+  const adminLibraryIds: string[] = [adminLibrary1, adminLibrary2, adminLibrary3, adminLibrary4];
+  const adminModerateLibraryIds: string[] = [adminModerateLibrary1, adminModerateLibrary2, adminModerateLibrary3, adminModerateLibrary4];
+  const user1LibraryIds: string[] = [user1Library1, user1Library2, user1Library3, user1Library4, user1Library5];
+  const siteRoles: string[] = [
+    Site.RoleEnum.SiteManager,
+    Site.RoleEnum.SiteContributor,
+    Site.RoleEnum.SiteCollaborator,
+    Site.RoleEnum.SiteConsumer,
+    Site.RoleEnum.SiteManager
+  ];
+  const user2LibraryIds: string[] = [user2Library1, user2Library2, user2Library3, user2Library4];
+  const user2LibraryIdsDelete: string[] = [user2Library5Delete, user2Library6Delete, user2Library7Delete];
+
+  let adminSitesApi: SitesApi;
+  let user2SitesApi: SitesApi;
+
+  test.beforeAll(async () => {
+    test.setTimeout(120000);
+    try {
+      await apiClientFactory.setUpAcaBackend('admin');
+      await apiClientFactory.createUser({ username: username1 });
+      await apiClientFactory.createUser({ username: username2 });
+
+      adminSitesApi = await SitesApi.initialize('admin');
+      for (const libraryId of adminLibraryIds) {
+        await adminSitesApi.createSite(libraryId);
+      }
+
+      for (const libraryId of adminModerateLibraryIds) {
+        await adminSitesApi.createSite(libraryId, Site.VisibilityEnum.MODERATED);
+      }
+      await adminSitesApi.addSiteMember(adminLibrary3, username2, Site.RoleEnum.SiteContributor);
+
+      const user1SitesApi = await SitesApi.initialize(username1, username1);
+      for (let i = 0; i < user1LibraryIds.length; i++) {
+        await user1SitesApi.createSite(user1LibraryIds[i]);
+        await user1SitesApi.addSiteMember(user1LibraryIds[i], username2, siteRoles[i]);
+      }
+
+      user2SitesApi = await SitesApi.initialize(username2, username2);
+      for (const libraryId of user2LibraryIds) {
+        await user2SitesApi.createSite(libraryId);
+      }
+
+      for (const libraryId of user2LibraryIdsDelete) {
+        await user2SitesApi.createSite(libraryId);
+      }
+
+      await new Promise((resolve) => setTimeout(resolve, timeouts.extraLarge));
```

Comment:
Is there maybe a method to pause or wait for some time within Playwright that we could use instead of `setTimeout`?

Reply:
beforeAll do not allow to use wait methods that are based on context or page:
Error: "context" and "page" fixtures are not supported in "beforeAll" since they are created on a per-test basis.